code/modules/power/apc.dm

```diff
@@ -628,12 +628,13 @@
 /obj/machinery/power/apc/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, \
 										datum/tgui/master_ui = null, datum/ui_state/state = default_state)
 	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
-	if(ui)
-		ui.set_autoupdate(state = (failure_timer ? 1 : 0))
+
 	if(!ui)
 		ui = new(user, src, ui_key, "apc", name, 535, 515, master_ui, state)
 		ui.open()
 		ui.set_autoupdate(state = (failure_timer ? 1 : 0))
+	else
```

Comment:
this was totally not my point.

see how you've reduced it from 3 lines to... 3 lines, as opposed to 2 how I said?
take the entire:
```
if(ui)
    ui.set_autoupdate(state = (failure_timer ? 1 : 0))
```
and put it down here, now it covers BOTH if the ui does and doesn't exist's cases, 2 lines, not 3.